### PR TITLE
Introduce custom errors classes

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

Related: https://github.com/mcollina/undici/issues/160

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -393,6 +393,25 @@ Calls [`client.close(callback)`](#close) on all the clients.
...
+Undici exposes a variety of error objects that you can use to enhance your error handling.
+You can find all the error objects inside the `errors` key.
+
+```js
+const { errors } = require('undici')
+```
+
+| Error                  | Error Codes              | Description                                            |
+| ---------------------- |--------------------------|--------------------------------------------------------|
+| `ConfigurationError`   |  `UND_ERR_CONFIGURATION` | Generated when in case of a bad configuration.         |
```
Comment (ronag) [inline README.md:408]: Would prefer invalid argument error?

File: lib/errors.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    super(message)
+    this.name = 'UndiciError'
+    this.code = 'UND_ERR'
+  }
+}
+
+class TimeoutError extends UndiciError {
+  constructor (message) {
+    super(message)
+    Error.captureStackTrace(this, TimeoutError)
```
Comment (ronag) [inline lib/errors.js:14]: I’m sure there is a reason for this. I just don’t know it. 😅

File: README.md [hunk 3]
```diff
@@ -393,6 +393,25 @@ Calls [`client.close(callback)`](#close) on all the clients.
...
+| `TimeoutError`         |  `UND_ERR_TIMEOUT`       | Generated when a request exceeds the `timeout` option. |
+| `RequestAbortedError`  |  `UND_ERR_ABORTED`       | Generated in case of an aborted request.               |
```
Comment (ronag) [inline README.md:410]: I guess this should only be if the user aborted the event?

Server aborts I guess falls under socket error?

File: lib/errors.js (hunk 2 above)
Comment (delvedor) [reply lib/errors.js:14]: https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt :)

**Comment by ronag** [comment]
Landed in f679a0655d7e2e2153e0e556b738faf9f02f3850

**Comment by mcollina** [comment]
I likely prefer to use the "squash" button on Github vs the node core way. It's way simpler to maintain.

**Pull request closed without merging**